src/core/plugin/Plugin.h

```diff
@@ -116,6 +119,8 @@ class Plugin final {
     /// Get Plugin from lua engine
     static auto getPluginFromLua(lua_State* lua) -> Plugin*;
 
+    constexpr friend auto tieer(Plugin const& plugin) { return std::tie(plugin.name, plugin.version, plugin.path); }
```

Comment:
I think it's more readable to make the return type of this explicit. I think it's probably something like:
```c++
using key_type = std::tuple<const std::string&, const std::string&, const fs::path&>;
```

I would also rename this to something more descriptive like `getKeyTuple()` and add a description:
```c++
/// Returns a tuple that is suitable for identifying this plugin (e.g., to be used in a map, set, or sorted vector).
```

Lastly, to avoid unnecessary comparisons, we should use the most common differentiator (the path) as the first component of the tuple.

Reply:
Oh, then again, changing the order of the tuple will also modify the behavior of `<`. Can you clarify the reasoning behind the current ordering?